### DISCUSSION TITLE
Database tables of custom feed server template

### DISCUSSION THREAD

**Discussion by venteto** [MarshalX/atproto#275] [Questions] [answered]
In your feedgen template repo's [database.py file](https://github.com/MarshalX/bluesky-feed-generator/blob/main/server/database.py), what are the `SubscriptionState` and `DbMetadata` tables for? If everything in a new feed generator app had all references to either of these tables completley removed, what bad things would happen over time? Can either one of them be removed safely or no?

Why is updating the cursor or the database version necessary?

I ask because I'm going to attempt to handle migrations in an offline Django app instead so I can manage additional new tables that way and do easy backup of a remote Postgres database, without having to figure out how to reimplement all of that in Flask. Then I just want to leave copies of the essential models in the `database.py` file that Flask needs to actually save indexed posts and serve multiple feeds.

Thanks for any insights.

**Comment by MarshalX** [comment, marked as answer]
Saving the cursor to the database is required for data backfilling. For example, your server was down for 3 hours. And you want to continue from the moment when you stopped. You are getting a cursor from DB and continue work. There are plenty of scenarios including problems from the relay side (which could be an outage too). You should create an analog table using Django ORM which will keep cursors for firehose consumers. 

The meta table is the same as the Django migrations table. It was needed for migrating cursors between Firehouse URIs. For smooth migration from the old state of the repo. You can drop this table.

**Reply by venteto** [reply]
This is good to know, thanks
